packages/validator/docs/conditions.md

```diff
@@ -0,0 +1,443 @@
+# Conditions
+
+## Basic condition
+
+Conditions consist of `__if` and `__then` objects, these objects are placed into array `__conditions`, which can contain any amount of condition objects. Object `__if` contains parameter name with regular expression, if the regular expression is matched in provided specification, validator will check if specification matches everything that's in the `__then` object.
+
+Even if evaluation of `__then` object takes place, all messages from this validation are discarded and replaced with single error message.
+
+### Template
+
+```json
+{
+  "numbers": {
+    "__arrayItem": {
+      "value": {},
+      "description": {},
+      "__conditions": [
+        {
+          "__if": {
+            "value": "^one$"
+          },
+          "__then": {
+            "description": {
+              "__regexp": "^This is required by one$",
+              "__catch": {
+                "__level": "error"
+              }
+            }
+          }
+        },
+        {
+          "__if": {
+            "value": "^two$"
+          },
+          "__then": {
+            "description": {
+              "__regexp": "^This is required by two$",
+              "__catch": {
+                "__level": "error"
+              }
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+---
+### Valid specification
```

Comment:
It is missing intentionally, as output of successful validation is specified in the basics, since it looks the same everywhere, here it would be just redundant and make the readme much longer without providing any value.